File: Decorator/Java/CoffeeShowDecorator/src/CoffeeShopDecorator/ChocolateCondiment.java
package CoffeeShopDecorator;

//ConcreteDecorator
public class ChocolateCondiment extends CondimentDecorator
{
    public ChocolateCondiment(Beverage beverage) 
    {
    	super(beverage);
        description = "chocolate";
    }

    @Override
    public double getPrice()
    {
        return wrappedBeverage.getPrice() + 0.2d;
    }

    @Override
    public String getDescription()
    {
        return String.format("%s, %s", wrappedBeverage.getDescription(), description);
    }
}

File: Decorator/C#/CoffeeShopDecorator/ChocolateCondiment.cs
namespace CoffeeShopDecorator
{
    //ConcreteDecorator
    public class ChocolateCondiment : CondimentDecorator
    {
        public ChocolateCondiment(Beverage beverage) : base(beverage)
        {
            Description = "chocolate";
        }

        public override string GetDescription()
        {
            return $"{base.GetDescription()}, {Description}";
        }

        public override decimal GetPrice()
        {
            return base.GetPrice() + 0.2m;
        }
    }
}